src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/LanguageServerFeatureOptions.cs

```diff
@@ -14,5 +14,17 @@ internal abstract class LanguageServerFeatureOptions
         public abstract string HtmlVirtualDocumentSuffix { get; }
 
         public abstract bool SingleServerCompletionSupport { get; }
+
+        public string GetRazorCSharpFilePath(string razorFilePath) => razorFilePath + CSharpVirtualDocumentSuffix;
+        
+        public string GetRazorHtmlFilePath(string razorFilePath) => razorFilePath + HtmlVirtualDocumentSuffix;
+
+        public string GetRazorFilePath(string filePath)
+        {
+            filePath = filePath.Replace(CSharpVirtualDocumentSuffix, string.Empty);
+            filePath = filePath.Replace(HtmlVirtualDocumentSuffix, string.Empty);
```

Comment:
I guess you were already doing this, but could this break if users name their files weird things like "Component.g.cs.razor"?

Reply:
yup, totally could! 😄